content/articles/support-vector-regression-in-python/index.md

```diff
@@ -0,0 +1,303 @@
+    && (W^Tx  + b)-y_i \ \le \ \epsilon + \zeta^{*}_i\\
+    &&\zeta^ \ \ge0, \ \zeta^*\ge0
+\end{aligned}
+$
+Above is a linear optimization problem and the [Lagrangian multipliers](https://machinelearningmastery.com/a-gentle-introduction-to-method-of-lagrange-multipliers/) is the appropriate technique for solving such a problem.
+
+Since we clearly understood how the Support Vector Regression works and how to formulate its optimization problem, let's now implement this algorithm in python.
+
+### Implementation of Support Vector Regression in Python
+As in any other implementation, first, we get the necessary libraries in place. The code below imports these libraries.
+
+```python
+# get the libraries
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+
+```
+Once our libraries are imported, we need to read the data we will work on to our working space.
+
+```python
+# get the dataset
+dataset = pd.read_csv('/content/drive/MyDrive/Position_Salaries.csv')
+# our dataset in this implementation is small, and thus we can print it all instead of viewing only the end
+print(dataset)
+
+```
+### Output:
+
+The above dataset contains ten instances. A significant feature in the dataset above is the `Level` column. The `Position` column is just a description of the `Level` column, and thus it adds no value in our analysis. With this said, we shall separate our data into the feature and a test set. As discussed above, we have only one feature in this dataset, and therefore, we shall carry out our feature-study variable separation as shown in the code below.
+
+```python
+# split the data into featutes and target variable seperately
+X_l = dataset.iloc[:, 1:-1].values # features set
+y_p = dataset.iloc[:, -1].values # set of study variable
+
+```
+We can look at our feature set using the `print()` function.
+
+```python
+print(X_l)
+```
+Output:
+```bash
+[[ 1]
+ [ 2]
+ [ 3]
+ [ 4]
+ [ 5]
+ [ 6]
+ [ 7]
+ [ 8]
+ [ 9]
+ [10]]
+```
+From this output, it's clear that the X_l variable is a 2D array. Similarly, we can have a look at the y_p variable.
+
+```python
+print(y_p)
+```
+Output:
+![output](/engineering-education/support-vector-regression-in-python/y-vriable.png)
+
+It's seen from the output above that the y_p variable is a vector, i.e., a 1D array. Also, we need to note that the values of y_p are huge compared to x_l. Therefore, if we implement a model on this data, the study variable will dominate the feature variable, and therefore its contribution to the model will be neglected. Therefore, later we will need to scale it to the same range as the scaled features set so that no variable dominates the other. The challenge is that the StandardScaler class we use to scale our data expects a 2D array; otherwise, it throws an error. Due to this, we have to reshape our y_p variable to be in 2D. The code below reshapes our study variable.
```

Comment:
Making them as bulleted points would look better and make it easier for the reader to read.